pmc/journal_recommendation/square_bins_word_ratios.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Calculate Odds Ratios for each Square Bin
+
+# In[1]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+from collections import Counter
+import csv
+import json
+from pathlib import Path
+import re
+
+import numpy as np
+import pandas as pd
+import spacy
+from tqdm import tqdm_notebook
+
+from annorxiver_modules.corpora_comparison_helper import (
+    aggregate_word_counts,
+    get_term_statistics
+)
+
+
+# # Gather Paper Bins Dataframe
+
+# In[2]:
+
+
+pmc_df = pd.read_csv(
+    "output/paper_dataset/paper_dataset_tsne_square.tsv",
+    sep="\t"
+)
+pmc_df.head()
+
+
+# In[3]:
+
+
+word_count_folder = Path("../pmc_corpus/pmc_word_counts/")
+
+
+# In[4]:
+
+
+bin_group = pmc_df.groupby("squarebin_id")
+
+
+# In[5]:
+
+
+spacy_nlp = spacy.load('en_core_web_sm')
+stop_word_list = list(spacy_nlp.Defaults.stop_words)
+
+
+# In[6]:
+
+
+global_word_counter = Counter()
+for name, group in tqdm_notebook(bin_group):
+    files = [
+        f"{word_count_folder.resolve()}/{doc}.tsv"
+        for doc in group.document.tolist()
+    ]
+    
+    agg_word_count = aggregate_word_counts(files, disable_progressbar=True)
+    
+    filtered_agg_word_count = {
+       term[0]:agg_word_count[term] 
+        for term in agg_word_count 
+        if term[1] != 'SPACE' and term[0] not in stop_word_list
+    }
+    
+    global_word_counter.update(Counter(filtered_agg_word_count))
```

Comment:
Is there a reason to cast the `dict` to a `Counter` before passing it into update?

Reply:
No hard reason for it being casted as a Counter. Just helps orient me programming wise to know that I'm working with a counter rather than pure dictionary.